### PR TITLE
Adds a possibility to delete a talk

### PR DESCRIPTION
**Pull request by heiglandreas**
This PR introduces an ability to delete a talk in web2. It can be used by every user that is capable of editing a talk. 

It adds a "Delete Talk" button beneath the "Edit Talk" button and implements the "delete"-endpoint.

### CODE REVIEW

File: app/src/Talk/TalkController.php [hunk 1]
```diff
@@ -479,6 +480,92 @@ public function unlinkSpeaker($eventSlug, $talkSlug, $username)
...
+        }
+
+        if (false === $talk) {
+            $this->application->notFound();
+            return;
+        }
+
+        // default values
+        $data              = [];
+        $data['talk_uri'] = $talk->getApiUri();
```
Comment (exussum12) [inline app/src/Talk/TalkController.php]: Really small but it looks like these were designed to line up ?

File: app/src/Talk/TalkController.php [hunk 2]
```diff
@@ -479,6 +480,92 @@ public function unlinkSpeaker($eventSlug, $talkSlug, $username)
...
+    public function deleteTalk($eventSlug, $talkSlug)
+    {
+        $thisUrl = $this->application->urlFor(
+            'talk-delete',
+            ['eventSlug' => $eventSlug, 'talkSlug' => $talkSlug]
+        );
+
+        if (! isset($_SESSION['user'])) {
+            $this->application->redirect(
+                $this->application->urlFor('not-allowed') . '?redirect=' . $thisUrl
```
Comment (exussum12) [inline app/src/Talk/TalkController.php]: Can this if statement be combined with the one below?

File: app/templates/Talk/_common/deleteform.html.twig [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                <p>
+                    The talk and all associated comments will be removed from
+                    the system. There will be no way of retrieving that
+                    information after you delete the talk!
+                    This can not be undone!
+                </p>
+            </div>
+        </div>
+        <div class="row">
+            <div class="col-sm-12">
```
Comment (exussum12) [inline app/templates/Talk/_common/deleteform.html.twig]: sm-12 is the default here ?

**Comment by liam-wiltshire** [comment]
@heiglandreas Becuase of some other recent updates (i.e. the talk_links work) - this now fails - foreign keys.

Either we need to:
- Delete the talk links before deleting the talk
- Update the delete talk API to automatically clear the talk_links
- Update the foreign key constraint to cascade the delete

Are you able to pick this up for me please? :-)

**Comment by liam-wiltshire** [comment]
To confirm, if I delete a talk without any slides etc, then it works perfectly :)

**Comment by heiglandreas** [comment]
Sure! I was already wondering about the push to my branch :)

I'd vote for option 2 - Updating the delete talk API to automatically clear the talk_links

**Comment by liam-wiltshire** [comment]
Haha, was just resolving the conflict for you - all part of the service ;-).

That's fine by me - thanks

**Comment by heiglandreas** [comment]
Oh! Thank you very much, sir! 😄  Much appreciated!

**Pull request merged**